xilem_core/src/sequence.rs

```diff
@@ -524,6 +524,83 @@ where
     }
 }
 
+impl<State, Action, Context, Element, Marker, Seq, Message, const N: usize>
+    ViewSequence<State, Action, Context, Element, Vec<Marker>, Message> for [Seq; N]
+where
+    Seq: ViewSequence<State, Action, Context, Element, Marker, Message>,
+    Context: ViewPathTracker,
+    Element: ViewElement,
+{
+    type SeqState = [Seq::SeqState; N];
+
+    #[doc(hidden)]
+    fn seq_build(&self, ctx: &mut Context, elements: &mut AppendVec<Element>) -> Self::SeqState {
+        // there's no enumerate directly on an array
+        let mut idx = 0;
+        self.each_ref().map(|vs| {
+            let state = ctx.with_id(ViewId::new(idx), |ctx| vs.seq_build(ctx, elements));
+            idx += 1;
+            state
+        })
+    }
+
+    #[doc(hidden)]
+    fn seq_rebuild(
+        &self,
+        prev: &Self,
+        seq_state: &mut Self::SeqState,
+        ctx: &mut Context,
+        elements: &mut impl ElementSplice<Element>,
+    ) {
+        for (gen, ((seq, prev_seq), state)) in self.iter().zip(prev).zip(seq_state).enumerate() {
+            ctx.with_id(
+                ViewId::new(gen.try_into().expect("usize should fit in u64")),
+                |ctx| {
+                    seq.seq_rebuild(prev_seq, state, ctx, elements);
+                },
+            );
+        }
+    }
+
+    #[doc(hidden)]
+    fn seq_message(
+        &self,
+        seq_state: &mut Self::SeqState,
+        id_path: &[ViewId],
+        message: Message,
+        app_state: &mut State,
+    ) -> MessageResult<Action, Message> {
+        let (start, rest) = id_path
+            .split_first()
+            .expect("Id path has elements for [ViewSequence; N]");
+
+        let index: usize = start
+            .routing_id()
+            .try_into()
+            .expect("u64 should fit in usize");
```

Comment:
This can just be an unwrap, as it's actually impossible (the index was previously created from a usize)